scripts/lib/add-on.sh

```diff
@@ -69,7 +82,13 @@ function install_add_on() {
   fi
 
   echo "installing addon $new_addon_version"
-  aws eks create-addon $ENDPOINT_FLAG --cluster-name "$CLUSTER_NAME" --addon-name $VPC_CNI_ADDON_NAME --resolve-conflicts OVERWRITE --addon-version $new_addon_version --region "$REGION"
+
+  # If installing the addon after deleting it once, we should retain the Service Account Role ARN of the Addon
+  if [ "$EXISTING_SERVICE_ACCOUNT_ROLE_ARN" != "null" ]; then
+     SA_ROLE_ARN_ARG="--service-account-role-arn $EXISTING_SERVICE_ACCOUNT_ROLE_ARN"
+  fi
+
+  aws eks create-addon $ENDPOINT_FLAG $SA_ROLE_ARN_ARG --cluster-name "$CLUSTER_NAME" --addon-name $VPC_CNI_ADDON_NAME --resolve-conflicts OVERWRITE --addon-version $new_addon_version --region "$REGION"
```

Comment:
if `$EXISTING_SERVICE_ACCOUNT_ROLE_ARN` is null then for the create-addon we would pass `$SA_ROLE_ARN_ARG` null, is that fine?

Reply:
I think $SA_ROLE_ARN_ARG would be an empty string which I believe should be fine, but I have not tested it :D

Reply:
right, it would be empty. Should be fine then.

Reply:
Yeah should count as an extra space